moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1835,12 +1838,32 @@ bool PlanningScene::processCollisionObjectMove(const moveit_msgs::CollisionObjec
     }
 
     collision_detection::World::ObjectConstPtr obj = world_->getObject(object.id);
+
+    // Update subframe poses
+    moveit::core::FixedTransformsMap subframe_poses = obj->subframe_poses_;
+    if (!object.subframe_poses.empty())
+    {
+      Eigen::Isometry3d frame_pose;
+      for (std::size_t i = 0; i < object.subframe_poses.size(); ++i)
+      {
+        tf2::fromMsg(object.subframe_poses[i], frame_pose);
+        std::string name = object.subframe_names[i];
+        subframe_poses[name] = t * frame_pose;
+      }
+    }
+    else if (!obj->subframe_poses_.empty())  // If there were subframes but no new poses are given
+    {
+      ROS_WARN_NAMED(LOGNAME, "Object '%s' had subframes, but no new ones were given for Move operation. ",
+                     object.id.c_str());
+    }
+
```

Comment:
i am not happy with the need to have the names vector transmitted here, but there's no way around that.
at least, it should be a normal use case of this to specify no subframes, even if the world object includes some.